lib/extractor/layer.ts

```diff
@@ -53,6 +55,73 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
+  ociArchiveFilesystemPath: string,
+  extractActions: ExtractAction[],
+): Promise<{
+  layers: ExtractedLayers[];
+  manifest: OciArchiveManifest;
+}> {
+  return new Promise((resolve, reject) => {
+    const tarExtractor: Extract = extract();
+
+    const layers: Record<string, ExtractedLayers> = {};
+    const manifests: Record<string, OciArchiveManifest> = {};
+    let imageIndex;
+
+    tarExtractor.on("entry", async (header, stream, next) => {
+      if (header.type === "file") {
+        if (isImageIndexFile(header.name)) {
+          imageIndex = await streamToJson<OciImageIndex>(stream);
+        } else {
+          const streamCopy1 = new PassThrough();
```

Comment:
not critical at all - I prefer to change `streamCopy1` to `jsonStream` and `streamCopy2` to `layerStream`

Reply:
done!